test/fixtures/with-props.js

```diff
@@ -1,8 +1,11 @@
 import { styled } from 'astroturf';
-import withProps from 'recompose/withProps';
 
-const RedPasswordInput = withProps({
+const RedPasswordInput = styled('input').attrs({ type: 'password' })`
```

Comment:
Am I understanding this correctly to think that, e.g.:

```js
const Foo = styled('foo')`bar`;

const Foo1 = Foo.attrs(...);
const Foo2 = Foo.attrs(...);
```

wouldn't work?

In which case this seems like a bit of an odd API. Couldn't `styled` just take an optional second argument?

Reply:
Yes that's right, there is one real benefit tho to this style, it matches styled components so tooling understands it. Specifically syntax highlighting extensions.

TBF tho your example wouldn't work in SC either, attrs() is on the template tag not the component.

But yeah it's not how I would do this API but there's an annoying precedent set already :/

Reply:
ohh, right, the API doesn't work that way. okay